### PR TITLE
gtk-vnc 0.5.4 (new formula)

### PR DESCRIPTION
**Pull request by vicamo**
Signed-off-by: You-Sheng Yang (楊有勝) <vicamo@gmail.com>

### CODE REVIEW

File: gtk-vnc.rb [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+  depends_on "gdk-pixbuf"
+  depends_on "gnutls"
+  depends_on "gtk+3"
+  depends_on "intltool" => :build
+  depends_on "libtool" => :build
+  depends_on "pkg-config" => :build
+  depends_on "pulseaudio" => :optional
+  depends_on "vala" => :optional
+
+  patch :DATA
```
Comment (mikemcquaid) [inline gtk-vnc.rb]: Can you please submit this patch to the upstream developers of this project and add a link to the upstream patch submission and explanation of why the patch is needed in a comment in the formula file. Thanks!

**Comment by vicamo** [comment]
Pushed new commit with patch removed.

Upstream has already filed a bug https://bugzilla.gnome.org/show_bug.cgi?id=602371 for build failures gtk-vnc 0.3 on Mac OS X 10.6.2. The patch I had before was carried from gtk-vnc 0.5.3 compiled with Xcode 5 or 6. It turns out that's no longer necessary for Xcode 7.

File: gtk-vnc.rb [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    ENV.append "CPPFLAGS", "-D_DARWIN_C_SOURCE"
+
+    system "./configure", *args
+
+    system "make"
+    system "make", "install"
+  end
+
+  test do
+    File.exist? "#{bin}/gvncviewer"
```
Comment (mikemcquaid) [inline gtk-vnc.rb]: Can this test be modified to do something more substantial than e.g. `--version` or `--help`? See `cmake.rb` for an example of an application formula with a good test and `tinyxml2.rb` for an example of a library formula with a good test. Thanks!

**Comment by vicamo** [comment]
Push new commit to address previous comment. gtk-vnc is shipped with only two examples that links gtk-vnc itself, so it's already an exmple to prove validness of built libgtk-vnc-2.0. Therefore I simply put `gvncviewer --help-all` here to ensure it's runnable.

File: gtk-vnc.rb (hunk 2 above)
Comment (vicamo) [inline gtk-vnc.rb]: pushed a new commit to run `gvncviewer --help-all` instead.

**Comment by mikemcquaid** [comment]
:+1: from me but can't pull at the moment so will let another maintainer do so.

**Comment by xu-cheng** [comment]
Thanks for the pull request! 🎉 Homebrew depends on contributions from community members like you and we're grateful for your support.

**Comment by xu-cheng** [comment]
Sorry the bottle job seems to fail to run.

@BerwTestBot test this please

**Comment by vicamo** [comment]
@xu-cheng is there any err message that I can have a reference?

**Comment by xu-cheng** [comment]
Sorry, but you put the formula file in the wrong place. It should be put inside `Library/Formula` directory.

**Comment by vicamo** [comment]
My apologies. Moved into `Library/Formula/gtk-vnc.rb`.

**Comment by vicamo** [comment]
Rebase and use `args = %W[...]` style.

**Comment by vicamo** [comment]
Address depends_on grouping style commented https://github.com/Homebrew/homebrew/pull/48860#discussion_r52092079 in all my pull requests.

**Comment by vicamo** [comment]
Fix missing dep to libgcrypt.

**Comment by vicamo** [comment]
Carry previous patch to build against Xcode < 7.2.

**Comment by apjanke** [comment]
@BrewTestBot test this please

**Comment by mikemcquaid** [comment]
Thanks for your contribution to Homebrew! Without people like you submitting PRs we couldn't run this project. You rock!

**Pull request closed without merging**